plugins/parquet/parquetembed.cpp

```diff
@@ -0,0 +1,1886 @@
+
+            RtlDynamicRowBuilder rowBuilder(m_resultAllocator);
+            const RtlTypeInfo *typeInfo = m_resultAllocator->queryOutputMeta()->queryTypeInfo();
+            assertex(typeInfo);
+            RtlFieldStrInfo dummyField("<row>", NULL, typeInfo);
+            size32_t len = typeInfo->build(rowBuilder, 0, &dummyField, pRowBuilder);
+            return rowBuilder.finalizeRowClear(len);
+        }
+        else
+            failx("Error processing result row");
+    }
+    return nullptr;
+}
+
+void ParquetRowStream::stop()
+{
+    m_resultAllocator.clear();
+    m_shouldRead = false;
+}
+
+void ParquetRowBuilder::xpathOrName(StringBuffer &outXPath, const RtlFieldInfo *field) const
+{
+    outXPath.clear();
+
+    if (field->xpath)
+    {
+        if (field->xpath[0] == xpathCompoundSeparatorChar)
+        {
+            outXPath.append(field->xpath + 1);
+        }
+        else
+        {
+            const char *sep = strchr(field->xpath, xpathCompoundSeparatorChar);
+
+            if (!sep)
+            {
+                outXPath.append(field->xpath);
+            }
+            else
+            {
+                outXPath.append(field->xpath, 0, static_cast<size32_t>(sep - field->xpath));
+            }
+        }
+    }
+    else
+    {
+        outXPath.append(field->name);
+    }
+}
+
+/**
+ * @brief Gets a Boolean result for an ECL Row
+ *
+ * @param field Holds the value of the field.
+ * @return bool Returns the boolean value from the result row.
+ */
+bool ParquetRowBuilder::getBooleanResult(const RtlFieldInfo *field)
+{
+    nextField(field);
+
+    if ((*array_visitor)->type == 0)
+    {
+        NullFieldProcessor p(field);
+        return p.boolResult;
+    }
+    if ((*array_visitor)->type != 1)
+    {
+        failx("Incorrect type for field %s.", field->name);
+    }
+    auto i = !m_pathStack.empty() && m_pathStack.back().nodeType == CPNTSet ? m_pathStack.back().childrenProcessed++ : currentRow;
+    return (*array_visitor)->bool_arr->Value(i);
+}
+
+/**
+ * @brief Gets a data result from the result row and passes it back to engine through result.
+ *
+ * @param field Holds the value of the field.
+ * @param len Length of the Data value.
+ * @param result Used for returning the result to the caller.
+ */
+void ParquetRowBuilder::getDataResult(const RtlFieldInfo *field, size32_t &len, void *&result)
+{
+    nextField(field);
+
+    if ((*array_visitor)->type == 0)
+    {
+        NullFieldProcessor p(field);
+        rtlUtf8ToDataX(len, result, p.resultChars, p.stringResult);
+        return;
+    }
+    if ((*array_visitor)->type == 6)
+    {
+        auto i = !m_pathStack.empty() && m_pathStack.back().nodeType == CPNTSet ? m_pathStack.back().childrenProcessed++ : currentRow;
+        auto view = (*array_visitor)->large_bin_arr->GetView(i);
+        rtlUtf8ToDataX(len, result, rtlUtf8Length(view.size(), view.data()), view.data());
```

Comment:
That would probably work, but I wasn't able to find the function rtlDataToDataX in the platform code. I tried with rtlDataToData and it couldn't copy to the target because it is a null pointer, which I assume is the difference between non-X copy methods and X copy methods.